src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/datatype/ComplexTypes.java

```diff
@@ -0,0 +1,73 @@
+package io.stargate.sgv2.jsonapi.api.model.command.table.definition.datatype;
+
+import io.stargate.sgv2.jsonapi.service.schema.tables.ApiDataType;
+import io.stargate.sgv2.jsonapi.service.schema.tables.ComplexApiDataType;
+import io.stargate.sgv2.jsonapi.service.schema.tables.PrimitiveApiDataType;
+
+/** Interface for complex column types like collections */
+public interface ComplexTypes extends ColumnType {
+
+  /** A map type implimentation */
+  public static class MapType implements ComplexTypes {
```

Comment:
Instead of inner-static class, shouldn't `MapType` be main-level types.
Or is there specific reason to enclose them in `ComplexTypes`?

Reply:
Resolved this.